libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -36,11 +36,9 @@ using namespace Linux;
 UARTDriver::UARTDriver(bool default_console) :
     device_path(NULL),
     _packetise(false),
-    _flow_control(FLOW_CONTROL_DISABLE)
+    _device{new ConsoleDevice()}
```

Comment:
we always call begin() for each uart. Why do you have to allocate it here rather than just do it in the begin() method?